ddtrace/contrib/kafka/patch.py

```diff
@@ -159,6 +164,17 @@ def traced_produce(func, instance, args, kwargs):
         rate = config.kafka.get_analytics_sample_rate()
         if rate is not None:
             span.set_tag(ANALYTICS_SAMPLE_RATE_KEY, rate)
+
+        if config.kafka.distributed_tracing_enabled:
+            # inject headers with Datadog tags:
+            try:
+                Propagator.inject(span.context, kwargs["headers"])
+            except KeyError:
+                try:
+                    Propagator.inject(span.context, args[6])
+                except IndexError:
```

Comment:
In what scenarios are both key and index errors raised? Can we use `get_argument_value` and `set_argument_value here: https://github.com/DataDog/dd-trace-py/blob/fd1c9e12478089b91db24a17ea2efa994feba5ac/ddtrace/internal/utils/__init__.py#L17?